machine/models/machine_tag.py

```diff
@@ -0,0 +1,31 @@
+from odoo import models, fields, api, _
+from odoo.exceptions import ValidationError
+
+from ..utils import generate_name_get
+
+
+class MachineTag(models.Model):
+    """Model to segment machines by tags."""
+
+    _name = 'machine.tag'
+    _description = 'Machine Instance Tag'
+
+    name = fields.Char(required=True)
+    parent_id = fields.Many2one('machine.tag', "Parent Tag")
+    color = fields.Integer(string='Color Index', default=10)
+
+    _sql_constraints = [
+        ('name_uniq', 'unique (name)', "Tag name already exists !"),
```

Comment:
No, tags here are not intended for such usage, it would just complicate tags. Might be implemented in a future if its really needed, but now I don't see a good reason to do that.